include/internal/refcount.h

```diff
@@ -25,12 +25,13 @@
 
 #   define HAVE_ATOMICS 1
 
-typedef _Atomic int CRYPTO_REF_COUNT;
+typedef struct {
+    _Atomic int val;
+} CRYPTO_REF_COUNT;
```

Comment:
Is there a reason this has to be changed to a struct? I currently don't see why the implementations that have atomic support would require a struct. I assume the compiler will actually generate the same code for both.

Reply:
I did this for a couple of reasons:
- There is one flavour that requires a lock and hence must use a structure.  Making all of the different versions use a structure maintains consistency.   I take advantage of this right down the bottom of the files, however changing those to work either way would be straightforward.
- Freedom to change things, it's an opaque struct so we can make changes in the future if required.

I agree that the compiler is going to generate the same code regardless.